src/Network/Network/VirtualNetworkGateway/NewAzureVirtualNetworkGatewayCommand.cs

```diff
@@ -260,6 +264,17 @@ public class NewAzureVirtualNetworkGatewayCommand : VirtualNetworkGatewayBaseCmd
         [ValidateNotNullOrEmpty]
         public string[] CustomRoute { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The generation for this VirtualNetwork VPN gateway. Must be None if GatewayType is not VPN.")]
+        [ValidateSet(
```

Comment:
It should be either `ValidateSet(a, b, c)` or `PSArgumentCompleter(a, b, c)`, not together. `PSArgumentCompleter` is preferable.